src/mongo/db/auth/external/cyrus_sasl_server_session.cpp

```diff
@@ -238,7 +241,7 @@ StatusWith<std::tuple<bool, std::string> > CyrusSASLServerSession::step(StringDa
 }
 
 StringData CyrusSASLServerSession::getPrincipalName() const {
-    const char* username;
+    const char* username = nullptr;
     int result = sasl_getprop(_saslConnection, SASL_USERNAME, (const void**)&username);
```

Comment:
In that [line](https://github.com/percona/percona-server-mongodb/pull/1001/files#diff-76b5162fc32296d3e51c36921e7d2ddc92bd8fed72375132ff020e6153d6e5a4R116), the C-style conversion was replaced with a `static_cast`. Is it worth using `static_cast`/`reinterpret_cast` in the other places in the file where old-fashioned conversions are still used?

Reply:
I updated PR. Unfortunately no way to use static_cast's. Due to ugly C libraries API I had to use const_cast and reinterpret_cast.